src/server/table.cc

```diff
@@ -57,37 +57,32 @@ SlotStats& SlotStats::operator+=(const SlotStats& o) {
   return *this;
 }
 
-void LockTable::Key::MakeOwned() const {
-  if (std::holds_alternative<std::string_view>(val_))
-    val_ = std::string{std::get<std::string_view>(val_)};
-}
-
 size_t LockTable::Size() const {
   return locks_.size();
 }
 
 std::optional<const IntentLock> LockTable::Find(std::string_view key) const {
   DCHECK_EQ(KeyLockArgs::GetLockKey(key), key);
 
-  if (auto it = locks_.find(Key{key}); it != locks_.end())
+  if (auto it = locks_.find(Key::FromView(key)); it != locks_.end())
     return it->second;
   return std::nullopt;
 }
 
 bool LockTable::Acquire(std::string_view key, IntentLock::Mode mode) {
   DCHECK_EQ(KeyLockArgs::GetLockKey(key), key);
 
-  auto [it, inserted] = locks_.try_emplace(Key{key});
-  if (!inserted)            // If more than one transaction refers to a key
-    it->first.MakeOwned();  // we must fall back to using a self-contained string
+  auto [it, inserted] = locks_.try_emplace(Key::FromView(key));
+  if (!inserted)                              // If more than one transaction refers to a key
+    const_cast<Key&>(it->first).MakeOwned();  // we must fall back to using a self-contained string
```

Comment:
fwiw it's exactly the same as before, except that now `MakeOwned()` is non-`const` (because it's not a `const` operation), and we do the `const_cast` because know that it wouldn't matter